### PR TITLE
simplify the selection of years in the training_pipeline

### PR DESCRIPTION
**Pull request by JialuJialu**

### CODE REVIEW

File: cities/modeling/modeling_utils.py [hunk 1]
```diff
@@ -91,49 +91,17 @@ def prep_wide_data_for_inference(
...
-    outcome_years_to_keep = [
-        year
-        for year in outcome.columns[2:]
-        if year_min <= int(year) <= year_max + forward_shift
-    ]
-
-    outcome_years_to_keep = [
-        year for year in outcome_years_to_keep if year in intervention.columns[2:]
-    ]
-
```
Comment (JialuJialu) [inline cities/modeling/modeling_utils.py:115]: The line 106 to line 114 works the same as 
`
outcome_years_to_keep = [year for year in outcome.columns[2:] if year in intervention.columns[2:]
`
It seems that the intent was to write 
`
outcome_years_to_keep = [year for year in outcome.columns[2:] if year - forwardshift in intervention.columns[2:]
`

File: cities/modeling/modeling_utils.py [hunk 2]
```diff
@@ -91,49 +91,17 @@ def prep_wide_data_for_inference(
...
-    outcome = outcome[outcome_years_to_keep]
-
-    # shift outcome `forward_shift` steps ahead
-    # for the prediction task
-    outcome_shifted = outcome.copy()
-
-    for i in range(len(outcome_years_to_keep) - forward_shift):
-        outcome_shifted.iloc[:, i] = outcome_shifted.iloc[:, i + forward_shift]
-
```
Comment (JialuJialu) [inline cities/modeling/modeling_utils.py:124]: The shifting could be completed in one step through renaming.

**Comment by JialuJialu** [comment]
I didn't get `error: Module "pyro.optim" has no attribute "Adam"` when running the check locally.

**Review by Niklewa — DISMISSED** [review]
The improvements to `modeling_utils.py` look good.

The only issue is a linting error that is causing the checks to fail: it would reformat `/home/runner/work/cities/cities/cities/__init__.py`

**Review by rfl-urbaniak — CHANGES_REQUESTED** [review]
Multiple tests fail as you removed a variable that is needed downstream. Also, changing the optimizer to abide by the lint is not recommended. It seems there is a bug in mypy where it doesn't seem to see things in torch.

https://github.com/pytorch/pytorch/issues/33757

So instead I'd recommend ignoring this line by `mypy` and reverting to the original optimizer.

If you feel like fixing this, please do, if your otherwise preoccupied, lmk, we can stall this PR.

File: cities/modeling/modeling_utils.py [hunk 3]
```diff
@@ -124,50 +124,25 @@ def prep_wide_data_for_inference(
...
-    assert all(intervention["GeoFIPS"] == outcome["GeoFIPS"])
-
+    # This is for the downstream variable
     outcome_years_to_keep = [
         year
         for year in outcome.columns[2:]
-        if year_min <= int(year) <= year_max + forward_shift
+        if str(int(year) - forward_shift) in intervention.columns[2:]
     ]
```
Comment (JialuJialu) [inline cities/modeling/modeling_utils.py:133]: I think this is what intended here but if you want to have the same behavior as before, change it to `outcome_years_to_keep = [year for year in outcome.columns[2:] if year in intervention.columns[2:]]`

Comment (rfl-urbaniak) [reply cities/modeling/modeling_utils.py:133]: Thanks!

File: cities/modeling/modeling_utils.py (hunk 1 above)
Comment (rfl-urbaniak) [reply cities/modeling/modeling_utils.py:149]: Alright, I think the issue was dropping a variable that was needed elsewhere.

**Review by rfl-urbaniak — APPROVED** [review]

**Pull request merged**